Первый коммит на проверку

dlfilin opened the pull request:
Первый PR

dlfilin committed:
Первый коммит на проверку

ilshat-abdulin commented on src/main/kotlin/Main.kt:
```diff
@@ -1,5 +1,16 @@
+
+data class Note(val name: String, val content: MutableList<String>)
+
+data class Archive(val name: String, val notes: MutableList<Note>)
+
+val archives: MutableList<Archive> = mutableListOf()
```
🍏 не критично, но можно вынести в отдельный файл

ilshat-abdulin commented on src/main/kotlin/ScreenManager.kt:
```diff
@@ -0,0 +1,61 @@
+
+        when (currentScreen.screenID) {
+            1 -> {
+                when (i) {
+                    0 -> isExitPressed = true
+                    1 -> {
+                        println("Введите название нового архива:")
+                        archives.add(Archive(currentScreen.getTextInput(), mutableListOf()))
+                    }
+                    else -> {
+                        currentArchive = archives[i - 2]
+                        currentScreen = notesListScreen
+                    }
+                }
```
⚠ Видно, что имеется схожее поведение при том или ином screenId. Попробуй сделать ScreenMenu абстрактным классом  для наследования, чтобы создать конкретные экраны. Т.е., например, ArchiveMenu, NotesMenu, NoteMenu. Общую логику объединить в ScreenMenu, а остальное в конкретных реализациях

dlfilin committed:
2й коммит на проверку:
- сделал DataModel файл с дата классами и архивами
- Переименовал MenuScreen в Menu, сделал абстрактным, общую логику оставил
- Сделал 3 наследника от Menu, вынес специф.логику туда
- порефакторил немного код, убрал лишнее
